src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -617,6 +656,13 @@ BinderCheckTypeResult ReportInvalidOperation()
                             // Regular Addition
                             var leftResAdd = CheckTypeCore(errorContainer, node.Left, leftType, DType.Number, /* coerced: */ DType.String, DType.Boolean);
                             var rightResAdd = CheckTypeCore(errorContainer, node.Right, rightType, DType.Number, /* coerced: */ DType.String, DType.Boolean);
+                            
+                            // Deferred + number or number + Deferred
+                            if (leftKind == DKind.Deferred || rightKind == DKind.Deferred)
+                            {
```

Comment:
Is this missing in the other cases for leftKind? 
What about `Time(,...) + deferred)`?

Reply:
No, Time(,...) + deferred is not missed, we have test for that also.